pkg/util/enablementstatus.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import "fmt"
+
+// EnablementStatus can be 'enabled', 'disabled' or 'auto'.
+type EnablementStatus int
```

Comment:
With it's current location, name, and the fact that we're adding it as part of our public API `EnablementStatus` would appear to be somewhat vague.

I would be more comfortable with making this `internal/` for now and putting it under a package like `controller/util` or `utils/controller` or something of that nature to be more specific that this is about controllers, and to not expose it as a usable package.